DuckDuckGo/FeatureFlagging/Model/FeatureFlagOverrides.swift

```diff
@@ -0,0 +1,132 @@
+//
+//  FeatureFlagOverrides.swift
+//
+//  Copyright © 2024 DuckDuckGo. All rights reserved.
+//
+//  Licensed under the Apache License, Version 2.0 (the "License");
+//  you may not use this file except in compliance with the License.
+//  You may obtain a copy of the License at
+//
+//  http://www.apache.org/licenses/LICENSE-2.0
+//
+//  Unless required by applicable law or agreed to in writing, software
+//  distributed under the License is distributed on an "AS IS" BASIS,
+//  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+//  See the License for the specific language governing permissions and
+//  limitations under the License.
+//
+
+import BrowserServicesKit
+import Foundation
+import Persistence
+
+protocol FeatureFlagOverridesPersistor {
+    func value(for flag: FeatureFlag) -> Bool?
+    func set(_ value: Bool?, for flag: FeatureFlag)
+}
+
+struct FeatureFlagOverridesUserDefaultsPersistor: FeatureFlagOverridesPersistor {
+    let keyValueStore: KeyValueStoring
+
+    func value(for flag: FeatureFlag) -> Bool? {
+        let key = key(for: flag)
+        return keyValueStore.object(forKey: key) as? Bool
+    }
+
+    func set(_ value: Bool?, for flag: FeatureFlag) {
+        let key = key(for: flag)
+        keyValueStore.set(value, forKey: key)
+    }
+
+    private func key(for flag: FeatureFlag) -> String {
+        return "local-override.\(flag.rawValue)"
```

Comment:
There's one thing I forgot to mention about user defaults in shared containers.

There's an advantage in using keys that can be matched to a property name 1:1.  That will allow us to get KVO across processes for free on those user defaults (meaning if process A changes the value, process B will be notified).

In practical terms it would only require changing `local-override.enforceRoutes` to something like `localOverrideMyFeature`.

Would `localOverride`\(flag.rawValue.capitalized)` be acceptable?

Reply:
Yes, it absolutely would! Thanks for pointing that out, @diegoreymendez! This isn't the final implementation so I'll adjust it before the code is ready for review.